Add documentation changes for shallow snapshot v2 (#8207)
opensearch-project/documentation-website: _tuning-your-cluster/availability-and-recovery/remote-store/snapshot-interoperability.md

```
{% include copy-curl.html %}

> Once enabled, all requests using the [Snapshot API]({{site.url}}{{site.baseurl}}/api-reference/snapshots/index/) will remain the same for all snapshots. Therefore, do not disable the shallow snapshot setting after it has been enabled since disabling the setting could affect data durability.

Reviewer: ```suggestion
Once enabled, all requests using the [Snapshot API]({{site.url}}{{site.baseurl}}/api-reference/snapshots/index/) will remain the same for all snapshots. Therefore, do not disable the shallow snapshot setting after it has been enabled because disabling the setting could affect data durability.
```

- All nodes in the cluster must use OpenSearch 2.10 or later to take advantage of shallow copy snapshots.
- The `incremental` file count and size between the current snapshot and the last snapshot is `0` when using shallow copy snapshots.
- Searchable snapshots are not supported inside shallow copy snapshots.

> # Shallow Snapshot v2 (OpenSearch 2.17+)

Reviewer 2: ```suggestion
## Shallow snapshot v2 
```

- All nodes in the cluster must use OpenSearch 2.10 or later to take advantage of shallow copy snapshots.
- The `incremental` file count and size between the current snapshot and the last snapshot is `0` when using shallow copy snapshots.
- Searchable snapshots are not supported inside shallow copy snapshots.

## Shallow snapshot v2 

> Starting with OpenSearch 2.17, the shallow snapshot feature has been improved with a new version called `shallow snapshot v2`. Shallow snapshot v2 aims to makes snapshot operations more efficient and scalable by introducing the following enhancements:

Reviewer: ```suggestion
Starting with OpenSearch 2.17, the shallow snapshot feature offers an improved version called `shallow snapshot v2`, which aims to makes snapshot operations more efficient and scalable by introducing the following enhancements:
```

> 

Reviewer: Global: Should all instances of "Shallow snapshot v2" be `shallow snapshot v2`?

> * Deterministic snapshot operations: Shallow snapshot v2 makes snapshot operations are more deterministic, ensuring consistent and predictable behavior.

Reviewer: ```suggestion
* Deterministic snapshot operations: Shallow snapshot v2 makes snapshot operations more deterministic, ensuring consistent and predictable behavior.
```

> * Minimized cluster state updates: Shallow snapshot v2  minimizes the number of cluster state updates required during snapshot operations, reducing overhead and improving performance.

Reviewer: ```suggestion
* Minimized cluster state updates: Shallow snapshot v2 minimizes the number of cluster state updates required during snapshot operations, reducing overhead and improving performance.
```

* Scalability: Shallow snapshot v2 allows snapshot operations to scale independently of the number of shards in the cluster, enabling better performance and efficiency for large datasets.

> Shallow snapshot v2 must be enabled seperately from shallow copies to receive these benefits.

Reviewer 2: ```suggestion
Shallow snapshot v2 must be enabled separately from shallow copies to receive these benefits.
```

> Shallow snapshot v2 must be enabled separately from shallow copies to receive these benefits.

Reviewer: Can we get rid of "to receive these benefits"?

> Shallow snapshot v2 must be enabled separately from shallow copies to receive these benefits.

Reviewer 2: ```suggestion
Shallow snapshot v2 must be enabled separately from shallow copies.
```

> ### Enabling shallow nnapshot v2

Reviewer: ```suggestion
### Enabling shallow snapshot v2
```

> To enable the shallow snapshot v2, enable the following repository settings:

Reviewer: ```suggestion
To enable shallow snapshot v2, enable the following repository settings:
```

- `remote_store_index_shallow_copy: true`
- `shallow_snapshot_v2: true`

> The following example creates a shallow snapshot v2 repository:

Reviewer: Can we insert the noun after "example"?

> The following example creates a shallow snapshot v2 repository:

Reviewer 2: ```suggestion
The following example request creates a shallow snapshot v2 repository:
```

```bash
PUT /_snapshot/snap_repo
{
"type": "s3",
"settings": {
"bucket": "test-bucket",
"base_path": "daily-snaps",
"remote_store_index_shallow_copy": true,
"shallow_snapshot_v2": true
}
}
```
{% include copy-curl.html %}

### Limitations 

> Shallow snapshots v2 have the following limitations:

Reviewer: ```suggestion
Shallow snapshot v2 has the following limitations:
```

> * Shallow snapshot v2 only works for remote-backed indexes.

Reviewer: "only works for" => "is only supported by/for"?

> * Shallow snapshot v2 only works for remote-backed indexes.

Reviewer 2: ```suggestion
* Shallow snapshot v2 only supported for remote-backed indexes.
```
